lib/rspec/matchers/expecteds_for_multiple_diffs.rb

```diff
@@ -62,11 +62,11 @@ def self.truncated(description)
       end
 
       def diffs(differ, actual)
-        @expected_list.map do |(expected, diff_label)|
+        @expected_list.inject("") do |output, (expected, diff_label)|
           diff = differ.diff(actual, expected)
-          next if diff.strip.empty?
-          "#{diff_label}#{diff}"
-        end.compact.join("\n")
+          next output if /\A[[:space:]]*\z/ === diff
```

Comment:
I don't think `[[:space]]` is supported in regexes on 1.8.7, so this is going to have to change to support 1.8.7, or you can define an alternate implementation for 1.8.7.  We often do that like so:

https://github.com/rspec/rspec-expectations/blob/a271631e9b8e1749be876dbcc65dbd7a6a40d2db/lib/rspec/matchers/built_in/match.rb#L77-L91